CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_ActivityChain.uc

```diff
@@ -331,10 +331,17 @@ function SetupComplications (XComGameState NewGameState)
 					}
 				}
 
-				`CI_Log("Adding Complication" @ ComplicationTemplate.DataName @ "at" @ ComplicationRoll $ "%");
+				if (ComplicationRoll > 0)
+				{
+					`CI_Log("Adding Complication" @ ComplicationTemplate.DataName @ "at" @ ComplicationRoll $ "%");
 				
-				ComplicationState = ComplicationTemplate.CreateInstanceFromTemplate(NewGameState, self, ComplicationRoll);
-				ComplicationRefs.AddItem(ComplicationState.GetReference());
+					ComplicationState = ComplicationTemplate.CreateInstanceFromTemplate(NewGameState, self, ComplicationRoll);
+					ComplicationRefs.AddItem(ComplicationState.GetReference());
+				}
+				else
+				{
+					`CI_Log("Adding Complication" @ ComplicationTemplate.DataName @ "but roll failed");
+				}
```

Comment:
Can you explain this change?

Reply:
Just a bug I found while testing. Previously, if a complication had a 50% chance to attach, and that roll failed, the complication would still be attached to the chain, but it would have a 0% trigger chance. Now, if the attach roll fails, the complication is discarded.